src/singletons.js

```diff
@@ -31,7 +23,38 @@ async function ERC1820Registry (funder) {
 
   await web3.eth.sendSignedTransaction(ERC1820_REGISTRY_DEPLOY_TX);
 
-  return ERC1820RegistryArtifact.at(ERC1820_REGISTRY_ADDRESS);
+  return getDeployedERC1820Registry();
+}
+
+async function getDeployedERC1820Registry () {
+  const environment = require('./config/environment').getEnviroment();
+
+  if (environment === 'truffle') {
+    let truffleContract;
+    [ '@truffle/contract', 'truffle-contract' ].forEach(pkg => {
+      if (truffleContract === undefined) {
+        truffleContract = tryRequire(pkg);
+      }
+    });
+
+    if (truffleContract === undefined) {
+      throw new Error(`
```

Comment:
There is a newline here. Alternatively put the <code>`</code> in the next line.
```suggestion
      throw new Error(`\
```